### PR TITLE
[LIST]

### PR DESCRIPTION
**Pull request by sfroment**
Fix splice func now reaction are more like c++ list plice

### CODE REVIEW

File: srcs/libc/list/src/s_list_splice.c [hunk 1]
```diff
@@ -30,99 +30,54 @@
...
-	cell_after = D_LIST_CELL(next)(end);
-	if (cell_before == NULL)
+	tmp = other_start;
+	count = 0;
+	while (my_start->v_data != other_end->v_data)
 	{
-		v_this->v_begin = cell_after;
-		if (cell_after != NULL)
-			cell_after->v_prev = NULL;
+		my_start->v_next = D_LIST_CELL(create)(my_start, tmp->v_next, tmp->v_data);
```
Comment (QuentinPerez) [inline srcs/libc/list/src/s_list_splice.c]: You must check if ```D_LIST_CELL(create)``` returns ```NULL`` and delete all the last cells allocated

Comment (sfroment) [inline srcs/libc/list/src/s_list_splice.c]: PR updated

File: srcs/libc/list/src/s_list_operator.c [hunk 2]
```diff
@@ -89,3 +89,18 @@ void		f_list_delete_if(t_list *v_this,
...
+bool		f_list_copy(t_list *v_this, t_list *from_list)
+{
+	t_list_cell	*cur;
+
+	D_LIST(clear)(v_this);
+	cur = from_list->v_begin;
+	while (cur != NULL)
+	{
+		if (D_LIST(push_back)(v_this, cur->v_data) == false)
+			return (false);
```
Comment (QuentinPerez) [inline srcs/libc/list/src/s_list_operator.c]: Clear the list, and restore v_this :)

Comment (sfroment) [inline srcs/libc/list/src/s_list_operator.c]: Shouldn't I do this in f_list_splice_empty ?

**Pull request closed without merging**